cgi-bin/LJ/Web.pm

```diff
@@ -1088,7 +1088,25 @@ sub create_url {
     my $proto = $opts{proto} // ( $opts{ssl} ? "https" : "http" );
     my $url = $proto . "://$host$path";
 
-    my $orig_args = $opts{cur_args} || DW::Request->get->get_args;
+    my $orig_args = $opts{cur_args} || DW::Request->get->get_args( preserve_case => 1 );
+
+    # Mimic previous behavior of lowercasing keys unless they start with Widget
+    # FIXME: this should eventually go away - see #1742
+    unless ( $opts{cur_args} ) {
+        foreach my $k ( keys %$orig_args ) {
+            next if $k eq lc $k;
+            next if $k =~ /^Widget\b/;
+            # already found another unintended side effect in /endpoints/draft
+            next if $k eq 'getProperties';
+
+            # don't complain about hand-typed arguments, but do convert them (for now)
+            my $referer = DW::Request->get->header_in("Referer");
+            warn "[#1742] lowercasing $k, url: $url (referer: $referer)\n"
```

Comment:
Ah oops. Yeah, I personally find code style of that form (lines almost look identical) to lead to line blindness and then I missed one. At my dayjob, we have a style guide preferring the use of Compounds Ifs like:

```next if $k eq lc $k || $k =~ /^Widget\b/ || $k eq 'getProperties';```